pkg/fga/client.go

```diff
@@ -0,0 +1,926 @@
+}
+
+// Resource types
+type ResourceType struct {
+	// Unique string ID of the resource type.
+	Type string `json:"type"`
+
+	// Set of relationships that subjects can have on resources of this type.
+	Relations map[string]interface{} `json:"relations"`
+}
+
+type ListResourceTypesOpts struct {
+	// Maximum number of records to return.
+	Limit int `url:"limit,omitempty"`
+
+	// The order in which to paginate records.
+	Order Order `url:"order,omitempty"`
+
+	// Pagination cursor to receive records before a provided ResourceType ID.
+	Before string `url:"before,omitempty"`
+
+	// Pagination cursor to receive records after a provided ResourceType ID.
+	After string `url:"after,omitempty"`
+}
+
+type ListResourceTypesResponse struct {
+	// List of Resource Types.
+	Data []ResourceType `json:"data"`
+
+	// Cursor pagination options.
+	ListMetadata common.ListMetadata `json:"list_metadata"`
+}
+
+type UpdateResourceTypeOpts struct {
+	// Unique string ID of the resource type.
+	Type string `json:"type"`
+
+	// Set of relationships that subjects can have on resources of this type.
+	Relations map[string]interface{} `json:"relations"`
+}
+
+// Warrants
+type Subject struct {
+	// The type of the subject.
+	ResourceType string `json:"resource_type"`
+
+	// The customer defined string identifier for this subject.
+	ResourceId string `json:"resource_id"`
+
+	// The relation of the subject.
+	Relation string `json:"relation,omitempty"`
+}
+
+type Warrant struct {
+	// Type of resource to assign a relation to. Must be an existing type.
+	ResourceType string `json:"resource_type"`
+
+	// Id of the resource to assign a relation to.
+	ResourceId string `json:"resource_id"`
+
+	// Relation to assign to the resource.
+	Relation string `json:"relation"`
+
+	// Subject of the warrant
+	Subject Subject `json:"subject"`
+
+	// Policy that must evaluate to true for warrant to be valid
+	Policy string `json:"policy,omitempty"`
+}
+
+type ListWarrantsOpts struct {
+	// Only return warrants whose resourceType matches this value.
+	ResourceType string `url:"resource_type,omitempty"`
+
+	// Only return warrants whose resourceId matches this value.
+	ResourceId string `url:"resource_id,omitempty"`
+
+	// Only return warrants whose relation matches this value.
+	Relation string `url:"relation,omitempty"`
+
+	// Only return warrants with a subject whose resourceType matches this value.
+	SubjectType string `url:"subject_type,omitempty"`
+
+	// Only return warrants with a subject whose resourceId matches this value.
+	SubjectId string `url:"subject_id,omitempty"`
+
+	// Only return warrants with a subject whose relation matches this value.
+	SubjectRelation string `url:"subject_relation,omitempty"`
+
+	// Maximum number of records to return.
+	Limit int `url:"limit,omitempty"`
+
+	// Pagination cursor to receive records after a provided Warrant ID.
+	After string `url:"after,omitempty"`
+
+	// Optional token to specify desired read consistency
+	WarrantToken string `url:"-"`
```

Comment:
Is it intentional to completely ignore this `WarrantToken` field when encoding this struct as query params?

Reply:
Ah, I see why now. It ends up being sent as a request header. 👍